packages/mobile/src/identity/securityCode.ts

```diff
@@ -0,0 +1,103 @@
+import {
+  ActionableAttestation,
+  AttestationServiceSecurityCodeRequest,
+  AttestationsWrapper,
+} from '@celo/contractkit/lib/wrappers/Attestations'
+import { PhoneNumberHashDetails } from '@celo/contractkit/src/identity/odis/phone-number-identifier'
+import { Address } from '@celo/utils/src/address'
+import BigNumber from 'bignumber.js'
+import Logger from 'src/utils/Logger'
+
+const TAG = 'identity/securityCode'
+
+function hashAddressToSingleDigit(address: Address): number {
+  return new BigNumber(address.toLowerCase()).modulo(10).toNumber()
+}
+
+export function getSecurityCodePrefix(issuerAddress: Address) {
+  return `${hashAddressToSingleDigit(issuerAddress)}`
+}
+
+export function extractSecurityCodeWithPrefix(message: string) {
+  const matches = message.match('\\s(\\d{8})\\s')
+  if (matches && matches.length === 2) {
+    return matches[1]
+  }
+  return null
+}
+
+export async function getAttestationCodeForSecurityCode(
+  attestationsWrapper: AttestationsWrapper,
+  phoneHashDetails: PhoneNumberHashDetails,
+  account: string,
+  attestations: ActionableAttestation[],
+  securityCodeWithPrefix: string
+) {
+  const securityCodePrefix = parseInt(securityCodeWithPrefix[0], 10)
+  const lookupAttestations = attestations.filter(
+    (attestation) => hashAddressToSingleDigit(attestation.issuer) === securityCodePrefix
+  )
```

Comment:
any thoughts on caching this at the same time that cache ActionableAttestations rather than hashing+filtering all attestations each time someone wants to send a security code to a service?

considering the number of actionable attestations will be low, the time savings it probably not super significant. but seems like the most efficient way to implement.